src/Objects/MonitorManager.vala

```diff
@@ -0,0 +1,375 @@
+         */
+
+        Display.MonitorMode? largest_mode_in_use = null;
+        Display.MonitorMode largest_mode = modes.get (0);
+        foreach (var mode in modes) {
+            if (mode.is_current) {
+                if (largest_mode_in_use == null) {
+                    largest_mode_in_use = mode;
+                } else if (largest_mode_in_use.width < mode.width) {
+                    largest_mode_in_use = mode;
+                }
+            }
+
+            // If there is one compatible setting, stop computing the fallback
+            if (largest_mode_in_use == null) {
+                if (largest_mode == null) {
+                    largest_mode = mode;
+                } else if (largest_mode.width < mode.width) {
+                    largest_mode = mode;
+                }
+            }
+        }
+
+        if (largest_mode_in_use != null) {
+            clone_virtual_monitor.set_current_mode (largest_mode_in_use);
+        } else {
+            clone_virtual_monitor.set_current_mode (largest_mode);
+        }
+
+        virtual_monitors.clear ();
+        virtual_monitors.add (clone_virtual_monitor);
+
+        notify_property ("virtual-monitor-number");
+        notify_property ("is-mirrored");
+    }
+
+    public void disable_clone_mode () {
+        double max_scale = Utils.get_min_compatible_scale (monitors);
+        var new_virtual_monitors = new Gee.LinkedList<Display.VirtualMonitor> ();
+        foreach (var monitor in monitors) {
+            var single_virtual_monitor = new Display.VirtualMonitor ();
+            var preferred_mode = monitor.preferred_mode;
+            var current_mode = monitor.current_mode;
+            if (global_scale_required) {
+                single_virtual_monitor.scale = max_scale;
+                if (max_scale in preferred_mode.supported_scales) {
+                    current_mode.is_current = false;
+                    preferred_mode.is_current = true;
+                } else if (!(max_scale in current_mode.supported_scales)) {
+                    Display.MonitorMode? largest_mode = null;
+                    foreach (var mode in monitor.modes) {
+                        if (max_scale in mode.supported_scales) {
+                            if (largest_mode == null || mode.width > largest_mode.width) {
+                                largest_mode = mode;
+                            }
+                        }
+                    }
+
+                    current_mode.is_current = false;
+                    largest_mode.is_current = true;
+                }
+            } else {
+                current_mode.is_current = false;
+                preferred_mode.is_current = true;
+                single_virtual_monitor.scale = preferred_mode.preferred_scale;
+            }
+
+            single_virtual_monitor.monitors.add (monitor);
+            new_virtual_monitors.add (single_virtual_monitor);
+        }
+
+        new_virtual_monitors.get(0).primary = true;
```

Comment:
Check if `new_virtual_monitors` is not empty.
Also you can use the `[0]` syntax with Gee.

Reply:
new_virtual_monitors can't be empty